spec.html

```diff
@@ -42955,6 +42966,383 @@ <h1>Properties of _TypedArray_ Instances</h1>
+        </dl>
+        <p>The <dfn id="standard-base64-alphabet">standard base64 alphabet</dfn> is the String whose elements are the code units corresponding to every letter and number in the Unicode Basic Latin block along with *"+"* and *"/"*; that is, it is *"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/"*.</p>
+        <emu-alg>
+          1. Let _chunkLength_ be the length of _chunk_.
+          1. If _chunkLength_ is 2, then
+            1. Set _chunk_ to the string-concatenation of _chunk_ and *"AA"*.
+          1. Else if _chunkLength_ is 3, then
+            1. Set _chunk_ to the string-concatenation of _chunk_ and *"A"*.
+          1. Else,
+            1. Assert: _chunkLength_ is 4.
+          1. Let _byteSequence_ be the unique sequence of 3 bytes resulting from decoding _chunk_ as base64 (such that applying the base64 encoding specified in section 4 of <a href="https://datatracker.ietf.org/doc/html/rfc4648">RFC 4648</a> to _byteSequence_ would produce _chunk_).
+          1. Let _bytes_ be a List whose elements are the elements of _byteSequence_, in order.
+          1. If _chunkLength_ is 2, then
+            1. Assert: _throwOnExtraBits_ is present.
+            1. If _throwOnExtraBits_ is *true* and _bytes_[1] ≠ 0, then
+              1. Throw a *SyntaxError* exception.
+            1. Return « _bytes_[0] ».
+          1. Else if _chunkLength_ is 3, then
+            1. Assert: _throwOnExtraBits_ is present.
+            1. If _throwOnExtraBits_ is *true* and _bytes_[2] ≠ 0, then
+              1. Throw a *SyntaxError* exception.
+            1. Return « _bytes_[0], _bytes_[1] ».
+          1. Else,
+            1. Return _bytes_.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-frombase64" type="abstract operation">
+        <h1>
+          FromBase64 (
+            _string_: a string,
+            _alphabet_: *"base64"* or *"base64url"*,
+            _lastChunkHandling_: *"loose"*, *"strict"*, or *"stop-before-partial"*,
+            optional _maxLength_: a non-negative integer,
+          ): a Record with fields [[Read]] (an integral Number), [[Bytes]] (a List of byte values), and [[Error]] (either ~none~ or a *SyntaxError* object)
+        </h1>
+        <dl class="header">
+        </dl>
+        <emu-alg>
+          1. If _maxLength_ is not present, then
+            1. Let _maxLength_ be 2<sup>53</sup> - 1.
+            1. NOTE: Because the input is a string, the length of strings is limited to 2<sup>53</sup> - 1 characters, and the output requires no more bytes than the input has characters, this limit can never be reached. However, it is editorially convenient to use a finite value here.
+          1. NOTE: The order of validation and decoding in the algorithm below is not observable. Implementations are encouraged to perform them in whatever order is most efficient, possibly interleaving validation with decoding, as long as the behaviour is observably equivalent.
```

Comment:
```suggestion
          1. NOTE: The order of validation and decoding in the algorithm below is not observable. Implementations are encouraged to perform them in whatever order is most efficient, possibly interleaving validation with decoding.
```

I don't think that last part needs to be made explicit.